Plans 2023: Adjust media queries for collapsed sidebar

aneeshd16 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Fixes Automattic/martech#1470

## Proposed Changes

* If the sidebar is collapsed, for the sake of simplicity, we treat it as if there is no sidebar, and the media queries used for the signup step are used.
* Renamed these mixins to better convey their meaning:

|Old|New|
|---|---|
|`onboarding-2023-pricing-grid-plans-breakpoints`|`pricing-grid-breakpoints`|
|`pricing-grid-2023-plans-section-breakpoints`|`pricing-grid-breakpoints-with-sidebar`|
|`plan-features-layout-switcher-onboarding`|`plan-features-layout-switcher`|
|`plan-features-layout-switcher-onboarding-plans-section`|`plan-features-layout-switcher-with-sidebar`|

* Breakpoints to be tested:

|Screen width on Signup flow OR sidebar collapsed on /plans|Screen width with expanded sidebar on /plans|Result|
|---|---|---|
|1500px|1772px|Large, grid width 1480px|
|1340px|1612px|Medium, grid width 1320px|
|880px|1152px|Small, grid width 860px|

_**Note: This change does not correctly handle the plan comparison grid.**_

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to `/start/plans?flags=onboarding/2023-pricing-grid`
* For each of the screen widths in the table above, confirm that the grid's width matches the one in the table. Decrease the screen width by 1px and confirm that the width/layout changes.
* Adjust the screen width manually by dragging the browser window horizontally, and confirm that the layout does not break.
* Go to `/plans/{SITE_ID}?flags=onboarding/2023-pricing-grid` and repeat the above two steps.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Commit:
Plans 2023: Adjust media queries for collapsed sidebar

Commit:
Adds media queries for plan comparison grid

claudiucelfilip commented:
Going through the breakpoints, I found some issues with the grid widths on `/plans` don't match the table:

**with expanded sidebar**
* at breakpoint 1772px, the grid widths start from 1435px (vs. 1480px)
* at breakpoint 1612px, the grid widths start from 1275px (vs. 1320px)
* at breakpoint 1152px, the grid widths start from 815px (vs. 860px)

**with collapsed sidebar**
* at breakpoint 1500px, the grid widths start from 1399px (vs. 1480px)
* at breakpoint 1340px, the grid widths start from 1239px (vs. 1320px)
* at breakpoint 880px, the grid widths start from 795px (vs. 860px)

These don't look bad or anything (it's actually an improved use of space), but I just wanted to point them out.

There are some other visual regressions on the `/plans` page that might need to be addressed:
* there's a double border between the Plans in the plans grid.

<img width="340" alt="Screenshot 2023-02-08 at 16 25 34" src="https://user-images.githubusercontent.com/2749938/217557693-17450fc5-7533-49c6-afd1-e1444e3a895a.png">

* at 880px, with collapsed sidebar, the Price Comparison Cells show a border and a weird alignment for `Visitor stats`

<img width="340" alt="Screenshot 2023-02-08 at 16 33 26" src="https://user-images.githubusercontent.com/2749938/217559824-66c55b89-0adc-43f3-9f0e-4de36ce91f60.png">

* at 881px, with expanded sidebar, the Price Comparison Cells don't show the border between columns. There are also missing margin adjustments made for `Live chat support` (to group it with Customer support) and below the `Storage` (or any last features in group) value (to not touch the bottom border)

<img width="320" alt="Screenshot 2023-02-08 at 16 34 59" src="https://user-images.githubusercontent.com/2749938/217560519-db6a8121-a4f9-4c60-a4f0-531ab861458c.png">

I haven't seen any issues on the `/start/plans` page

Commit:
Fixed double border

Commit:
Fixes plan comparison grid

aneeshd16 commented:
Thanks for the review @claudiucelfilip!

> Going through the breakpoints, I found some issues with the grid widths on /plans don't match the table:

Since this isn't a breaking change, I'll tackle this as a part of Automattic/martech#1484.

> there's a double border between the Plans in the plans grid.

Fixed, turns out there was a redundant `border` rule. (https://github.com/Automattic/wp-calypso/pull/73081/commits/363d24d43fac67f5beca985e7d5775cc55fa3c4f)

> at 880px, with collapsed sidebar, the Price Comparison Cells show a border and a weird alignment for Visitor stats
at 881px, with expanded sidebar, the Price Comparison Cells don't show the border between columns. There are also missing margin adjustments made for Live chat support (to group it with Customer support) and below the Storage (or any last features in group) value (to not touch the bottom border)

All these were due to a `max-width` media query that did not take the collapsed sidebar into account: https://github.com/Automattic/wp-calypso/blob/da453b0bfd3326fcc46649d2c73fbbbc5a68f79a/client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx#L125

I have removed the `max-width` media query and added the appropriate resets in the `min-width` media query via https://github.com/Automattic/wp-calypso/pull/73081/commits/6747fe0aa55aa1df6929e0ab8f108a9ae9df5ad3.

Commit:
Fixes plan row styles

claudiucelfilip commented on client/my-sites/plan-features-2023-grid/plan-comparison-grid.tsx:
```diff
@@ -99,6 +99,22 @@ const Row = styled.div< { isHiddenInMobile?: boolean } >`
+		&:last-of-type {
+			display: flex;
+			padding-top: 0;
+			padding-bottom: 24px;
```
We'll probably need to move the `padding-bottom: 24px;` to `.plan-comparison-grid__header.plan-is-footer`, otherwise the border stops before the bottom.
<img width="252" alt="Screenshot 2023-02-09 at 13 23 15" src="https://user-images.githubusercontent.com/2749938/217799529-d60387b9-660d-4f33-ac43-cded3f654b0e.png">

Commit:
Fixed bottom padding for plan row

claudiucelfilip reviewed (approved):
LGTM
